torch/_inductor/fx_passes/post_grad.py

```diff
@@ -484,6 +491,107 @@ def is_valid_splitwithsizes_cat(match):
     return True
 
 
+def same_layout(node1: torch.fx.Node, node2: torch.fx.Node):
+    """True if two nodes have the same size/strides"""
+    val1 = node1.meta.get("val")
+    val2 = node2.meta.get("val")
+    return (
+        val1 is not None
+        and val2 is not None
+        and val1.size() == val2.size()
+        and val1.stride() == val2.stride()
```

Comment:
if you haven't gotten any errors, looks good, but I'm surprised you havnet run into fake tensors that error on `.stride()` yet (nested tensor, sparse)